libraries/default.rb

```diff
@@ -128,16 +128,21 @@ def resolve_hostname(ip)
       require 'resolv'
       hostf = Resolv::Hosts.new
       dns = Resolv::DNS.new
-      # Try and resolve hostname first using /etc/hosts, then use DNS
+
+      # Try and resolve hostname first using DNS, then /etc/hosts
       begin
-        hostname = hostf.getname(ip)
+        hname = node['fqdn']
       rescue
         begin
-          hostname = dns.getname(ip)
+          hname = dns.getname(h)
```

Comment:
Where does `h` come from?

Reply:
Also, `dns.getname()` returns an object, not a string. Use `hname.to_s` to convert to string https://ruby-doc.org/stdlib-2.5.1/libdoc/resolv/rdoc/Resolv/DNS.html#method-i-getname